Pulse.Core/EncryptedConnection.cs

```diff
@@ -0,0 +1,63 @@
+using System.Threading.Channels;
+using Pulse.Core.Connections;
+
+namespace Pulse.Core;
+
+internal class EncryptedConnection : IConnection
+{
+    private readonly IConnection actualConnection;
+    private readonly PacketEncryptor encryptor;
+    public ChannelReader<Packet> IncomingAudio { get; }
+
+    public EncryptedConnection(IConnection actualConnection, PacketEncryptor encryptor)
+    {
+        if (!encryptor.Ready)
+        {
+            throw new InvalidOperationException("Encryptor is not properly configured!");
+        }
+
+        this.actualConnection = actualConnection;
+        this.encryptor = encryptor;
+        IncomingAudio = new CryptoPacketChannelReader(
+            this.actualConnection.IncomingAudio,
+            encryptor
+        );
+    }
+
+    public async Task SendPacketAsync(Packet packet, CancellationToken cancellationToken)
+    {
+        var encryptedPacket = await encryptor.EncryptAsync(packet);
+        await actualConnection.SendPacketAsync(encryptedPacket, cancellationToken);
+    }
+
+    private class CryptoPacketChannelReader : ChannelReader<Packet>
+    {
+        private readonly ChannelReader<Packet> actualChannelReader;
+        private readonly PacketEncryptor encryptor;
+
+        public CryptoPacketChannelReader(
+            ChannelReader<Packet> actualChannelReader,
+            PacketEncryptor encryptor
+        )
```

Comment:
The accepted convention is to place the ending parenthesis of a method next to the last parameter, and not on a separate line (that also holds for method calls)